### DISCUSSION TITLE
Time Graph

### DISCUSSION THREAD

**Discussion by gabriela-sartori** [gampleman/elm-visualization#108] [Q&A] [answered]
Is there a graph for time periods? In a period of 24h, I'd like to draw session durations with start/end from different users. Would it be better to draw it myself because it is too specific or to reuse and adapt a graph from elm-visualization?

Edit: I found a cool example of what I am thinking. Ignore instrument colors at bottom and album black vertical bars.
![image](https://user-images.githubusercontent.com/6143964/101858028-89782000-3b47-11eb-858b-1bfe53150411.png)

**Comment by deciob** [comment, marked as answer]
elm-visalisation is quite low level and doesn't lock you into any specific graph. I would imagine that what you want to achieve could be done with some help from the Axis and the Scale modules. The Y axis in this case would have an ordinal scale and the X axis a continuous scale, more specifically a time scale or even a linear scale. The scales will then help you create and position the [svg rect elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/rect) for your graph. If you still have doubts, I might be able to help you out with a simple proof of concept this evening.

**Reply by gampleman** [reply]
(Also, if you do come up with a solution, feel free to share the code here for any future users with the same question. Thanks)

**Reply by gabriela-sartori** [reply]
I am using Scale.time from an example I find on elm-visualization site, but I would like to display names instead of numbers in Y, is that possible? I am trying atm, first contact with graphs.

![image](https://user-images.githubusercontent.com/6143964/101995846-ceb56280-3cab-11eb-9b12-4d54ba450a73.png)

https://pastebin.com/TC6qMdUg

**Reply by deciob** [reply]
@G4BB3R if you want to display names instead of numbers, you can use https://package.elm-lang.org/packages/gampleman/elm-visualization/latest/Axis#tickFormat, but I think you could do this differently. In place of lines you could use bars and the Y scale could be an ordinal one. Depending on time, I might be able to send you an alternative example this evening for you to look at.

**Comment by deciob** [comment]
@G4BB3R I have an example that hopefully resembles what you want: https://ellie-app.com/bNRBPyRZX5Fa1
Note that the tuple is now (xStart, xEnd, y).
If you also want the bars to overlap with different thickness (like the example you show), then easiest way that I can think of is to use scales with different paddingInner.
I hope this helps.
